src/java_bytecode/expr2java.cpp

```diff
@@ -509,9 +502,22 @@ std::string expr2javat::convert(
   else if(src.id()=="pod_constructor")
     return "pod_constructor";
   else if(src.id()==ID_virtual_function)
-    return convert_function(src, "VIRTUAL_FUNCTION", precedence=16);
+  {
+    return "VIRTUAL_FUNCTION(" +
+      id2string(src.get(ID_C_class)) +
+      "." +
+      id2string(src.get(ID_component_name)) +
+      ")";
+  }
   else if(src.id()==ID_java_string_literal)
-    return '"'+id2string(src.get(ID_value))+'"'; // Todo: add escaping as needed
+    return '"'+MetaString(src.get_string(ID_value))+'"';
+  else if(src.id()==ID_constant && (type.id()==ID_bool || type.id()==ID_c_bool))
```

Comment:
Comment just for future improvement: we should factor out these "to_string" functions for each language, as currently similar code appears in multiple places in the code base.